wemake_python_styleguide/violations/best_practices.py

```diff
@@ -591,3 +593,24 @@ def some():
     #: Error message shown to the user.
     error_template = 'Found nested import "{0}"'
     code = 435
+
+
+class RedundantForElseViolation(ASTViolation):
+    """
+    Forbids to use ``else`` with ``break`` in ``for`` loop.
+
+    Reasoning:
+        This rule will reduce complexity, improve readability,
+         and protect for possible errors.
+
+    Solution:
+        Forbid use ``else`` with ``break`` statement.
```

Comment:
The solution should be "Refactor your ``else`` case logic to be inside the ``for`` body".